### ISSUE TITLE
squareBarVisualizer.setPlayer(0) is not happening

### ISSUE THREAD

**Issue by fahadammar** [GautamChibde/android-audio-visualizer#53]
I want to set the player to 0 so whenever the music is played, the visualizer animation occurs. But it's not happening, my app is crashing.

Crash:
 java.lang.RuntimeException: Cannot initialize Visualizer engine, error: -3
 
 Need the solution ASAP !!
 
 Thankyou so much, looking forward to reply's

**Comment by marvpaul** [comment]
Check out this: https://stackoverflow.com/questions/6971567/android-visualizer-class-throwing-runtime-exception

**Comment by marvpaul** [comment]
@fahadammar yep, just had the exact same error as you had. I resolved it with asking for the permissions: 
 <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />
